pydra/engine/specs.py

```diff
@@ -90,23 +101,66 @@ def hash(self):
         """Compute a basic hash for any given set of fields."""
         from .helpers import hash_value, hash_function
 
+        # if inp_hash already calculated and nothing has changed, the old value can be used
+        if self.changed is False and self.inp_hash:
```

Comment:
is this `if` necessary. shouldn't you always run this check since you don't have control over external files.

Reply:
you mean, without using `self.changed` at all? i think you're right